app/src/main/java/com/jerboa/MainActivity.kt

```diff
@@ -168,14 +174,40 @@ class MainActivity : ComponentActivity() {
                             selectMode = it.arguments?.getBoolean("select")!!
                         )
                     }
-                    composable(route = "createPost") {
+                    composable(
+                        route = "createPost",
+                        // TODO:  Handle Image mimetype too
+                        deepLinks = listOf(
+                            navDeepLink { mimeType = "text/plain" },
+                            navDeepLink { mimeType = "image/*" },
+                        )
+                    ) {
+
+                        val context = LocalContext.current
+                        val activity = context.findActivity()
+                        val text = activity?.intent?.getStringExtra(Intent.EXTRA_TEXT) ?: ""
+                        val image =
+                            activity?.intent?.getParcelableExtra<Parcelable>(Intent.EXTRA_STREAM) as? Uri
+                        // url and body will be empty everytime except when there is EXTRA TEXT in the intent
+                        var url = ""
+                        var body = ""
+                        if (Patterns.WEB_URL.matcher(text).matches()) {
+                            url = text
+                        } else {
+                            body = text
```

Comment:
I'm in the process of testing this out, but from what I can tell, when using share via an android browser, it puts the title on the first line, and the URL on the second. Maybe you could do a line split for that case.

Also, I think adding _title would be better than _body, right?

Reply:
What test data are you working with? The text could be any plain text from a single word to large chunks of paragraphs. it has no guaranteed structure. in most cases users will be using 'share link' which will just send the url that's what is handled on line 194.

> Also, I think adding _title would be better than _body, right?

If someone is copying an entire article to share you wouldn't want to fill the title with it.

I'm not sure how you managed to get both title and url, you might be using a share button from a website that uses some api to invoke the sharesheet and send formatted data

Reply:
I'm using several android browser's share button, which to me seems like the most common use case. You're on a site, and you want to share it to lemmy. That to me seems much more likely than selecting random text to share as a body of a lemmy post.

But also, you could still differentiate the two, by counting lines and seeing if the 2nd line is a URL.

Reply:
Here's some findings:

- Chrome - only shares the URL.
- Via - Shares two lines, first one is title, 2nd is URL
- Puffin - only shares the URL
- Firefox - only URL

In that case, I'll say this is good, because via seems to be the only one that does the 2-line method.